_docker/control.rb

```diff
@@ -177,22 +160,25 @@ def block_wait_searchisko_started(supporting_services)
   #the docker url is taken from DOCKER_HOST env variable otherwise
   Docker.url = tasks[:docker] if tasks[:docker]
 
+  # Output the new docker-compose file with the modified ports
+  File.delete('docker-compose.yml') if File.exists?('docker-compose.yml')
+  File.write('docker-compose.yml', ERB.new(File.read('docker-compose.yml.erb')).result)
+
   if tasks[:decrypt]
     puts 'Decrypting...'
     modify_env
   end
 
-  if tasks[:kill_all] && File.exists?('docker-compose.yml')
-    puts 'Killing and removing docker services...'
-    execute_docker_compose :down
+  begin
+    Docker::Network.get("#{project_name}_default")
+    network_exists = true
+  rescue
+    network_exists = false
   end
 
-  if tasks[:set_ports]
-    puts 'Setting ports...'
-    set_ports
-    # Output the new docker-compose file with the modified ports
-    File.delete('docker-compose.yml') if File.exists?('docker-compose.yml')
-    File.write('docker-compose.yml', ERB.new(File.read('docker-compose.yml.erb')).result)
+  if tasks[:kill_all] && File.exists?('docker-compose.yml') && network_exists
```

Comment:
I think ideally we should unit test logic like this by pulling it into a function e.g something like:

`
if tasks[:kill_all]
 kill_current_environment(tasks)
end 
`

And then the kill_current_environment task can be unit tested.

If not that, then I think we should at least document the logic behind this with a comment e.g. we only run docker-compose down if there is a network because the network could only exist if docker-compose up has run previously.